include/materials/norsand.tcc

```diff
@@ -114,54 +116,26 @@ bool mpm::NorSand<Tdim>::compute_elastic_tensor() {
 
 //! Compute stress invariants
 template <unsigned Tdim>
-Eigen::Matrix<double, 6, 1> mpm::NorSand<Tdim>::compute_stress_invariants(
+double* mpm::NorSand<Tdim>::compute_stress_invariants(
```

Comment:
For an in-out parameter, pass by pointer

Reply:
Perhaps what @kks32 suggests is this:
```suggestion
void mpm::NorSand<Tdim>::compute_stress_invariants(double* p, double* q, double* lode_angle, double* m_theta)
```
This is pass by pointer: https://www.geeksforgeeks.org/passing-by-pointer-vs-passing-by-reference-in-c/

Reply:
You can also pass by reference, both are ok I think :)

Reply:
Our general guidelines and ISO-C++ is that if you want to modify pass it as a pointer, if you want to not modify pass it as a `const &`. This will apply both for normal variables and smart pointers